coordinate/coordinator.go

```diff
@@ -0,0 +1,301 @@
+
+// ErrorUnableToStopCluster - can't stop the cluster
+var ErrorUnableToStopCluster = errors.New("Unable to stop cluster")
+
+const (
+	rootDir    = "/running/"
+	config     = "/config"
+	defaultTTL = 1200
+)
+
+// WatchFunc is a type that is used to handle responses from watches that are
+// created against etcd
+type WatchFunc func(*etcd.Response)
+
+// Client wraps the go-etcd package and its interactions with etcd into simple
+// operations that the flotilla-client and flotilla-server can use to spin up a
+// test cluster. the flotilla-client manages the cluster and watches for
+// for flotilla-server instances to join the flota specific cluster.
+type Client struct {
+	client     *etcd.Client
+	flota      string
+	id         string
+	watchLock  sync.Mutex
+	watchList  map[string]WatchFunc
+	stopList   map[string]chan bool
+	Registered bool
+}
+
+// getOrCreateRoot will delete the root + flota and its children if they do exist
+func (c *Client) getOrCreateRoot() error {
+
+	resp, err := c.client.Get(rootDir+c.flota, false, false)
+	if err != nil {
+		return nil
+	}
+
+	if resp == nil {
+		_, er := c.client.CreateDir(rootDir+c.flota, defaultTTL)
+		if er != nil {
+			return er
+		}
+		log.Println("Create new run ", rootDir+c.flota)
+
+	} else {
+
+		_, der := c.client.Delete(rootDir+c.flota, true)
+		if der != nil {
+			return der
+		}
+		log.Println("Remove previous run ", rootDir+c.flota)
+
+		_, cer := c.client.CreateDir(rootDir+c.flota, defaultTTL)
+		if cer != nil {
+			return cer
+		}
+		log.Println("Create new run ", rootDir+c.flota)
+	}
+
+	return nil
+
+}
+
+// StartCluster will
+// 1. create the flota dir that all the daemons will register to.
+// 2. wait until it all the daemons have been registered
+// 3. will timeout on its wait and return false or return true if all found
+//
+// These should only be used by the flotilla client
+func (c *Client) StartCluster(numdaemons int, startsleep int) (bool, error) {
+
+	err := c.getOrCreateRoot()
+	if err != nil {
+		c.StopCluster()
+		panic(err)
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	// Wait for the cluster to spin up for the sleep time or return when
+	// the cluster is all up. It checks every second.
+	step := 0
+
+	for {
+
+		step++
+		count := c.ClusterCount()
+		log.Printf("%d daemons have registered in %d seconds", count, step)
+
+		if count >= numdaemons {
+			return true, nil
+		}
+
+		if step > startsleep {
+			for _, member := range c.ClusterMembers() {
+				log.Printf("Member %s had joined", member)
+			}
+			return false, errors.New("Daemons not all registered")
+		}
+
+		time.Sleep(time.Second * 1)
+	}
+
+	return false, errors.New("Unable to start Cluter")
+}
+
+// StopCluster will close all watcher go routines and do a recursive delete on
+// rootDir + flota. This should only be used by the flotilla client
+func (c *Client) StopCluster() {
+	c.Close()
+	resp, err := c.client.Delete(rootDir+c.flota, false)
```

Comment:
I think I'll kill the stop right now, it does not do what it should do and 'stop' is a little final.